src/auto/functions.rs

```diff
@@ -643,15 +643,15 @@ pub fn intern_string<'a, P: Into<Option<&'a str>>>(string: P) -> Option<String>
     }
 }
 
-//pub fn io_add_watch<P: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(channel: /*Ignored*/&IOChannel, condition: /*Ignored*/IOCondition, func: /*Unknown conversion*//*Unimplemented*/IOFunc, user_data: P) -> u32 {
+//pub fn io_add_watch<P: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(channel: /*Ignored*/&IOChannel, condition: IOCondition, func: /*Unknown conversion*//*Unimplemented*/IOFunc, user_data: P) -> u32 {
 //    unsafe { TODO: call ffi::g_io_add_watch() }
 //}
 
-//pub fn io_add_watch_full<P: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(channel: /*Ignored*/&IOChannel, priority: i32, condition: /*Ignored*/IOCondition, func: /*Unknown conversion*//*Unimplemented*/IOFunc, user_data: P, notify: /*Unknown conversion*//*Unimplemented*/DestroyNotify) -> u32 {
+//pub fn io_add_watch_full<P: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(channel: /*Ignored*/&IOChannel, priority: i32, condition: IOCondition, func: /*Unknown conversion*//*Unimplemented*/IOFunc, user_data: P, notify: /*Unknown conversion*//*Unimplemented*/DestroyNotify) -> u32 {
 //    unsafe { TODO: call ffi::g_io_add_watch_full() }
 //}
 
-//pub fn io_create_watch(channel: /*Ignored*/&IOChannel, condition: /*Ignored*/IOCondition) -> Option<Source> {
+//pub fn io_create_watch(channel: /*Ignored*/&IOChannel, condition: IOCondition) -> Option<Source> {
```

Comment:
Still not ready for IOChannel?

Reply:
Maybe, but even if we were this would have to be manual so that the callback, etc can be set on the Source. See glib/src/source.rs for some other examples.